ovirt_imageio/tool/options.py

```diff
@@ -0,0 +1,90 @@
+# ovirt-imageio
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+"""
+Tool options.
+"""
+
+from .. _internal.units import KiB, MiB, GiB, TiB
+
+import argparse
+
+
+class Parser:
+
+    def __init__(self):
+        self._parser = argparse.ArgumentParser(
+            description="Transfer disk images")
+        self._parser.set_defaults(
+            command=lambda _: self._parser.print_help())
+        self._commands = self._parser.add_subparsers(title="commands")
+
+    def add_sub_command(self, name, help, func):
+        cmd = self._commands.add_parser(name, help=help)
+        cmd.set_defaults(command=func)
+
+        cmd.add_argument(
+            "--engine-url",
+            help="ovirt-engine URL.")
+
+        cmd.add_argument(
+            "--username",
+            help="ovirt-engine username.")
```

Comment:
The help shows this, --engine-url and --cafile) as optional. But are they really? The default username seems to be "None@N/A", which is unlikely to work, I assume. The default engine-url doesn't work either. Is there a default cafile?

Reply:
I suppose once --config/-c is implemented it will be really optional

Reply:
Yes, that will be cool. I wonder if the parser is able to display that properly. Maybe like this:
`{--config CONFIG_FILE, --engine-url ENGINE_URL --username USERNAME --cafile CAFILE [--password-file PASSWORD_FILE]}`

Reply:
Yes once we have the config everything is optional, but we are missing validation step
after parsing arguments, since engine url/username/password are really required to connect to engine.

We don't have a way to show that an argument is required via command line or config file, and we don't want to write our own parser for this.